pkg/reconciler/revision/resources/deploy_test.go

```diff
@@ -711,6 +1004,79 @@ func TestMakePodSpec(t *testing.T) {
 					withEnvVar("SERVING_SERVICE", "svc"),
 				),
 			}),
+	}, {
+		name: "complex pod spec for multiple containers with container data only to serving containers",
+		rev: revision("bar", "foo",
+			withContainers([]corev1.Container{
+				{
+					Name:  servingContainerName,
+					Image: "busybox",
+					Ports: []corev1.ContainerPort{{
+						ContainerPort: 8888,
+					}},
+				}, {
+					Name:  sidecarContainerName,
+					Image: "ubuntu",
+				}}),
+			withContainerConcurrency(1),
+			WithContainerStatuses([]v1.ContainerStatuses{{
+				ImageDigest: "busybox@sha256:deadbeef",
+			}, {
+				ImageDigest: "ubuntu@sha256:deadbffe",
+			}}),
+			func(revision *v1.Revision) {
+				revision.ObjectMeta.Labels = map[string]string{
+					serving.ConfigurationLabelKey: "cfg",
+					serving.ServiceLabelKey:       "svc",
+				}
+				revision.Spec.GetContainer().Command = []string{"/bin/bash"}
+				revision.Spec.GetContainer().Args = []string{"-c", "echo Hello world"}
+				container(revision.Spec.GetContainer(),
+					withTCPReadinessProbe(),
+					withEnvVar("FOO", "bar"),
+					withEnvVar("BAZ", "blah"),
+				)
+				revision.Spec.GetContainer().Resources = corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse("666Mi"),
+						corev1.ResourceCPU:    resource.MustParse("666m"),
+					},
+					Limits: corev1.ResourceList{
+						corev1.ResourceMemory: resource.MustParse("888Mi"),
+						corev1.ResourceCPU:    resource.MustParse("888m"),
+					},
+				}
+				revision.Spec.GetContainer().TerminationMessagePolicy = corev1.TerminationMessageReadFile
```

Comment:
Same comment as above, why not add all these to the container specification itself?